orchestrator/services/processes.py

```diff
@@ -682,7 +682,7 @@ def update_awaiting_process_progress(
     ensure_correct_callback_token(pstat, token=token)
 
     state = pstat.state.unwrap()
-    progress_key = state.get(DEFAULT_CALLBACK_PROGRESS_KEY, "callback_progress")
+    progress_key = state.get("__callback_progress_key", DEFAULT_CALLBACK_PROGRESS_KEY)
```

Comment:
This was probably inspired by the `callback_step()` which has a kwarg `result_key: str` that is stored in  the state as `"__callback_result_key": result_key`. I *think* this was added to be able to specify a different state key in case the `callback_result` key is needed for something else, but I'm not entirely sure.

Anyway, as far as I can tell there is currently no way to set `__callback_progress_key` in the state. Is this something you'd ever see becoming useful? If it isn't, I'd also be fine with simplifying this to:

```suggestion
    progress_key = DEFAULT_CALLBACK_PROGRESS_KEY
```

Reply:
We already weren't using the `__callback_result_key` functionality in the first place, so I have no strong preference here

Reply:
The initial reason for using this approach was mainly to replicate how it was done in the other callback step, which would give the workflow developer the ability to change the progress key. The developer would be able to set this by setting the `__callback_progress_key` in a previous step if I'm not mistaken.

But I agree, I don't see a real need for this so lets simplify it.